[quantization] Enabling inplace relu

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28710 [quantization] Enabling inplace relu**

Differential Revision: [D18146120](https://our.internmc.facebook.com/intern/diff/D18146120)

Commit:
[quantization] Enabling inplace relu

jerryzh168 commented:
does this work? why do we put this in the beginning?

z-a-f commented:
> does this work? why do we put this assert in the beginning?

Because we didn't support inplace initially, and there was a similar assertion in the functional.relu. After enabling the inplace, the assertion was never lifted in the module (only in the functional)

vadimkantorov commented on torch/nn/quantized/modules/activation.py:
```diff
@@ -31,7 +31,6 @@ class ReLU(torch.nn.ReLU):
-        assert not inplace, 'torch.nn.quantized.ReLU does not support inplace'
 
     def forward(self, input):
         return torch.nn.quantized.functional.relu(input)
```
should `inplace = self.inplace` be then passed to `torch.nn.quantized.functional.relu` call?

z-a-f replied:
Good point! Missed that!

Commit:
Update on "[quantization] Enabling inplace relu"


Differential Revision: [D18146120](https://our.internmc.facebook.com/intern/diff/D18146120)

Commit:
Update on "[quantization] Enabling inplace relu"


Differential Revision: [D18146120](https://our.internmc.facebook.com/intern/diff/D18146120)

Commit:
Update on "[quantization] Enabling inplace relu"


Differential Revision: [D18146120](https://our.internmc.facebook.com/intern/diff/D18146120)

Commit:
Update on "[quantization] Enabling inplace relu"


Differential Revision: [D18146120](https://our.internmc.facebook.com/intern/diff/D18146120)
